### ISSUE TITLE
Search time

### ISSUE THREAD

**Issue by rnirujogi** [vdemichev/DiaNN#276]
Dear Vadim,

I am repeatedly experiencing with multiple versions of DIA-NN in processing direct-DIA searches for both dia-PASEF and DIA from Orbitrap platforms. The search usually takes 10 to 12 days when I enable Phospho as a variable modification. 

The current search of DIA-Phospho data for 11 .raw files as below,
11 files will be processed
[0:00] Loading FASTA C:\Raja\Database\Human_UniprotSP_Cano+Iso_052021.fasta
[0:18] Processing FASTA
[3:15] Assembling elution groups
[6:22] 60396734 precursors generated
[6:22] Gene names missing for some isoforms
[6:22] Library contains 20373 proteins, and 20155 genes
[6:30] Encoding peptides for spectra and RTs prediction
[9:52] Predicting spectra and IMs
[6931:35] Predicting RTs

I think prediction of spectra and IMs taking enormous amount of time. I am wondering if I am missing something or could you suggest to fix this issue.

Many thanks.
Best,
Raja

**Comment by vdemichev** [comment]
Hi Raja,

This happens because 60 million precursors is a lot. Consider specifying less variable modifications / missed cleavages. About prediction of spectra: you only need to do this once, can then reuse the .predicted.speclib file as a spectral library. Please only use DIA-NN 1.8, not earlier versions. 

Best,
Vadim

**Comment by rnirujogi** [comment]
Hi Vadim,

Thanks for clarifying and yes it make sense to reduce missed cleavages at least to 1, as for majority of the Phosphopeptides are indeed identifiable with missed cleavages. Nonetheless the current log is as below just for your info,

11 files will be processed
[0:00] Loading FASTA C:\Raja\Database\Human_UniprotSP_Cano+Iso_052021.fasta
[0:18] Processing FASTA
[3:15] Assembling elution groups
[6:22] 60396734 precursors generated
[6:22] Gene names missing for some isoforms
[6:22] Library contains 20373 proteins, and 20155 genes
[6:30] Encoding peptides for spectra and RTs prediction
[9:52] Predicting spectra and IMs
[6931:35] Predicting RTs
[8284:11] Decoding predicted spectra and IMs
[8286:00] Decoding RTs
[8290:47] Saving the library to C:\Raja\DIA-NN searches\PPM1H-PROTAC_dDIA.predicted.speclib
[8293:49] Initialising library

[8294:36] First pass: generating a spectral library from DIA data
[8294:36] File #1/11
[8294:36] Loading run C:\Raja\DIA-NN searches\PPM1H-PROTAC_dDIA\RN-KB_211224_PPM1H-PROTAC_TiO2_vDIA_01.raw
[8302:50] 56185098 library precursors are potentially detectable
[8302:58] Processing...
[8316:21] RT window set to 28.4079
[8316:21] Peak width: 4.616
[8316:21] Scan window radius set to 10
[8316:27] Recommended MS1 mass accuracy setting: 6.74767 ppm
[8455:47] Removing low confidence identifications
[8455:50] Searching PTM decoys
[8509:26] Removing interfering precursors
[8509:41] Training neural networks: 82947 targets, 110658 decoys
[8510:14] Number of IDs at 0.01 FDR: 41428
[8510:16] Calculating protein q-values
[8510:21] Number of genes identified at 1% FDR: 5686 (precursor-level), 5167 (protein-level) (inference performed using proteotypic peptides only)
[8510:23] Quantification
[8510:31] Precursors with monitored PTMs at 1% FDR: 11714 out of 14167
[8510:31] Unmodified precursors with monitored PTM sites at 1% FDR: 17631 out of 20126
[8510:32] Precursors with PTMs localised (when required) with > 90% confidence: 5632 out of 10400
[8510:38] Quantification information saved to C:\Raja\DIA-NN searches\PPM1H-PROTAC_dDIA/C__Raja_DIA-NN searches_PPM1H-PROTAC_dDIA_RN-KB_211224_PPM1H-PROTAC_TiO2_vDIA_01_raw.quant.


Thanks and regards,
Raja

**Comment by vdemichev** [comment]
Thanks, so the main time is taken by the in silico library generation. Please just re-use C:\Raja\DIA-NN searches\PPM1H-PROTAC_dDIA.predicted.speclib as a regular spectral library for further searches. The reason it takes so long is because of memnory fragmentation caused by Pytorch - happens when predicting for very large precursor numbers. This is an issue with Pytroch, cannot really do anything about it.

**Comment by rnirujogi** [comment]
Thanks, Vadim. Nonetheless the search gets completed will follow your suggestions as well.

Best,
Raja
